internal/metrics/prometheus.go

```diff
@@ -0,0 +1,159 @@
+package metrics
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// Recorder is the interface fir recording metrics on a certain operation.
+type Recorder interface {
+	// REVIEW: add URL of Konnect API as a label?
+	RecordKonnectEntityOperationSuccess(operationType KonnectEntityOperation, entityType string, duration time.Duration)
+	RecordKonnectEntityOperationFailure(operationType KonnectEntityOperation, entityType string, duration time.Duration, statusCode int)
+}
+
+type KonnectEntityOperation string
+
+const (
+	KonnectEntityOperationTypeKey                        = "operation_type"
+	KonnectEntityOperationCreate  KonnectEntityOperation = "create"
+	KonnectEnttiyOperationUpdate  KonnectEntityOperation = "update"
+	KonnectEntityOperationDelete  KonnectEntityOperation = "delete"
+
+	KonnectEntityTypeKey = "entity_type"
+
+	SuccessKey   = "success"
+	SuccessTrue  = "true"
+	SuccessFalse = "false"
+
+	StatusCodeKey = "status_code"
+)
+
+// metric names for konnect entity operations.
+const (
+	// REVIEW: define a Namespace `gateway_operator` for creating prometheus metrics here?
```

Comment:
Can you elaborate what you mean in here? What would this namespace be used for?

Reply:
`namespace` is a field in the metric opts used as prefix of metric names: https://prometheus.io/docs/practices/naming/. I found it may not be necessary so I did not set it in creating metrics.